lib/utils/docker.coffee

```diff
@@ -70,36 +70,44 @@ exports.appendOptions = (opts) ->
 	]
 
 exports.generateConnectOpts = generateConnectOpts = (opts) ->
-	connectOpts = {}
-	# Firsly need to decide between a local docker socket
-	# and a host available over a host:port combo
-	if opts.docker? and not opts.dockerHost?
-		# good, local docker socket
-		connectOpts.socketPath = opts.docker
-	else if opts.dockerHost? and not opts.docker?
-		# Good a host is provided, and local socket isn't
-		connectOpts.host = opts.dockerHost
-		connectOpts.port = opts.dockerPort || 2376
-	else if opts.docker? and opts.dockerHost?
-		# Both provided, no obvious way to continue
-		throw new Error("Both a local docker socket and docker host have been provided. Don't know how to continue.")
-	else
-		# None provided, assume default docker local socket
-		connectOpts.socketPath = '/var/run/docker.sock'
-
-	# Now need to check if the user wants to connect over TLS
-	# to the host
-
-	# If any are set...
-	if (opts.ca? or opts.cert? or opts.key?)
-		# but not all
-		if not (opts.ca? and opts.cert? and opts.key?)
-			throw new Error('You must provide a CA, certificate and key in order to use TLS')
-		connectOpts.ca = opts.ca
-		connectOpts.cert = opts.cert
-		connectOpts.key = opts.key
-
-	return connectOpts
+	Promise = require('bluebird')
+	fs = require('mz/fs')
+
+	Promise.try ->
+		connectOpts = {}
+		# Firsly need to decide between a local docker socket
+		# and a host available over a host:port combo
+		if opts.docker? and not opts.dockerHost?
+			# good, local docker socket
+			connectOpts.socketPath = opts.docker
+		else if opts.dockerHost? and not opts.docker?
+			# Good a host is provided, and local socket isn't
+			connectOpts.host = opts.dockerHost
+			connectOpts.port = opts.dockerPort || 2376
+		else if opts.docker? and opts.dockerHost?
+			# Both provided, no obvious way to continue
+			throw new Error("Both a local docker socket and docker host have been provided. Don't know how to continue.")
+		else
+			# None provided, assume default docker local socket
+			connectOpts.socketPath = '/var/run/docker.sock'
+
+		# Now need to check if the user wants to connect over TLS
+		# to the host
+
+		# If any are set...
+		if (opts.ca? or opts.cert? or opts.key?)
+			# but not all
+			if not (opts.ca? and opts.cert? and opts.key?)
+				throw new Error('You must provide a CA, certificate and key in order to use TLS')
+
+			return Promise.all([fs.readFile(opts.ca), fs.readFile(opts.cert), fs.readFile(opts.key)])
+				.spread (ca, cert, key) ->
+					connectOpts.ca = ca.toString()
+					connectOpts.cert = cert.toString()
+					connectOpts.key = key.toString()
+					return connectOpts
```

Comment:
I think you can slightly simplify this to:

```coffeescript
Promise.props(ca: fs.readFile(opts.ca), cert: fs.readFile(opts.cert), key: fs.readFile(opts.key))
.then (result) ->
  _.mapValues result, value -> value.toString()
```

or even

```coffeescript
Promise.props(_.pick(opts, ['ca', 'cert', 'key']).map(fs.readFile))
.then (result) ->
  _.mapValues(result, value -> value.toString())
```

(although maybe that last step is more terse than it is clear, up to you)